### DISCUSSION TITLE
ERROR 203 for PYSWMM simulation

### DISCUSSION THREAD

**Discussion by jerry11011** [pyswmm/pyswmm#402] [Q&A]
I'm having a problem for running .inp file simulation.  Error says its error type 203. i checked the report file, it says not enough value for [XSECT].  Then i checked out the actual value in the inp file, and the rows that caused the error had a value of IRREGULAR. Are there any solutions for this? 

![errorcode](https://user-images.githubusercontent.com/124729158/217350844-2fa0b311-e2f8-4c50-a310-5615503e9eb7.png)
![error203rpt](https://user-images.githubusercontent.com/124729158/217350832-d51f401f-c524-4fe1-8719-6c1f603fe976.png)
![errorirregular](https://user-images.githubusercontent.com/124729158/217350847-e216195f-70bd-4581-a8ac-58a7d7c6eb07.png)

**Comment by bemcdonnell** [comment]
@jerry11011, this is a SWMM error, not a PySWMM error.  I recommend looked at the SWMM User's manual to gain more understanding of the INP structure.  One thing to consider is that pyswmm is currently pointing to SWMM-5.1.14, not SWMM-5.2.2.  That might be the issue you're having as well.

**Reply by jerry11011** [reply]
thanks for the reply :) Oh, does that mean PYSWMM doesn't support the latest features of SWMM?

**Reply by bemcdonnell** [reply]
@jerry11011 we are a release or two behind at this very moment.  I have a pull request in at the moment that does cover the updates to swmm5.2.2 :-)

**Reply by jerry11011** [reply]
Appreciate your comments!!

**Comment by dickinsonre** [comment]
If you can just try opening the inp file in SWMM5 - it should give you some messages as well in the status report.

**Reply by jerry11011** [reply]
it has no problem opening the file in SWMM5.. only goes wrong when simulating in PYSWMM :(

**Reply by bemcdonnell** [reply]
@jerry11011 are you opening it and running it in regular SWMM? So that works? Which version is it?

**Reply by jerry11011** [reply]
oh mine was 5.2 like you mentioned
